Highlighting not working after `:TSInstallFromGrammar`

**Versions:**

Using neovim 0.5.0 on Windows 10 with tree-sitter-cli 0.20.0 and latest nvim-treesitter plugin version.

**Problem:**

After installing [tree-sitter-tlaplus](https://github.com/tlaplus-community/tree-sitter-tlaplus) with `:TSInstallFromGrammar` and verifying its workings with `:checkhealth`, I opened a .tla file and no syntax highlighting was present. Tree-sitter syntax highlighting works for other languages installed using `:TSInstall` (verified using `:TSBufDisable highlight` and `:TSBufEnable highlight` on various C++ files). `tree-sitter highlight` also produces highlighting for .tla files.

**Steps to reproduce:**

1. Clone https://github.com/tlaplus-community/tree-sitter-tlaplus to ~/source/tree-sitter-tlaplus
2. Add the following to your init.vim:
```
" Configure local install of tree-sitter-tlaplus
lua <<EOF
local parser_config = require "nvim-treesitter.parsers".get_parser_configs()
parser_config.tlaplus = {
  install_info = {
    url = "~/source/tree-sitter-tlaplus",
    files = {"src/parser.c", "src/scanner.cc"}
  },
  filetype = "tla",
}
EOF
```
3. Open the x64 Native Tools Command Prompt for VS 2019 and start nvim
4. Run `:TSInstallFromGrammar tlaplus` to generate, compile, and install the tlaplus tree-sitter grammar
5. Run `:checkhealth` to verify that no issues exist in the Installation, Parser/Features, and health#treesitter#check sections; in the Parser/Features section, ensure that tlaplus is marked as supporting highlighting
6. Open a .tla file in nvim, for example the following:
```tla
---- MODULE Test ----
op == IF "Hello, world!" THEN BOOLEAN ELSE TRUE
====
``` 
7. Note that no syntax highlighting is applied

**Expected Behavior:**

I expected the .tla file to be displayed in nvim with syntax highlighting.

**Output of `:checkhealth nvim_treesitter`**
```
health#nvim_treesitter#check
========================================================================
## Installation
  - OK: `tree-sitter` found  0.20.0 (e85a279cf29da1b08648e27214dda20a841e57c8) (parser generator, only needed for :       TSInstallFromGrammar)
  - OK: `node` found v16.0.0 (only needed for :TSInstallFromGrammar)
  - OK: `git` executable found.
  - OK: `cl` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl" }
  - OK: Neovim was compiled with tree-sitter runtime ABI version 13 (required >=13). Parsers must be compatible with      runtime ABI.

## Parser/Features H L F I
  - json           ✓ ✓ ✓ ✓ .
  - tlaplus        ✓ . . . .
  - cpp            ✓ ✓ ✓ ✓ ✓
  - rust           ✓ ✓ ✓ ✓ ✓
  - c              ✓ ✓ ✓ ✓ ✓
  - java           ✓ ✓ . ✓ ✓

 Legend: H[ighlight], L[ocals], F[olds], I[ndents]
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang}
```

**Output of `nvim --version`**
```
NVIM v0.5.0
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: C:/Program Files (x86)/Microsoft Visual Studio/2017/Enterprise/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe /DWIN32 /D_WINDOWS /W3 -DNVIM_TS_HAS_SET_MATCH_LIMIT /MD /Zi /O2 /Ob1 /DNDEBUG /W3 -D_CRT_SECURE_NO_WARNINGS -D_CRT_NONSTDC_NO_DEPRECATE -DWIN32 -D_WIN32_WINNT=0x0600 -DINCLUDE_GENERATED_DECLARATIONS -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -ID:/a/neovim/neovim/build/config -ID:/a/neovim/neovim/src -ID:/a/neovim/neovim/nvim-deps/usr/include -ID:/a/neovim/neovim/build/src/nvim/auto -ID:/a/neovim/neovim/build/include
Compiled by runneradmin@fv-az152-786

Features: -acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM\sysinit.vim"
  fall-back for $VIM: "C:/Program Files/nvim/share/nvim"

Run :checkhealth for more info
```

> Shouldn't the :TSInstallFromGrammar command do that automatically by copying from the tree-sitter-tlaplus/queries directory or is that not its intended function?

I think that was done before, but must upstream queries aren't neovim specific, so we require them to be in the repo now.

I have tried to install your grammar, but I got

```
nvim-treesitter[tlaplus]: Error during compilation                                                                                                                                                                                              
src/scanner.cc:169:12: error: character too large for enclosing character literal type                                                                                                                                                          
      case '∧': return LAND;                                                                                                                                                                                                                    
           ^                                                                                                                                                                                                                                    
src/scanner.cc:171:12: error: character too large for enclosing character literal type                                                                                                                                                          
      case '∨': return LOR;                                                                                                                                                                                                                     
           ^                                                                                                                                                                                                                                    
src/scanner.cc:176:12: error: character too large for enclosing character literal type                                                                                                                                                          
      case '〉': return RIGHT_DELIMITER;                                                                                                                                                                                                        
           ^                                                                                                                                                                                                                                    
src/scanner.cc:226:7: error: use of undeclared identifier 'memcpy'                                                                                                                                                                              
      memcpy(&buffer[offset], (char*)&alignment_column, copied);                                                                                                                                                                                
      ^                                                                                                                                                                                                                                         
src/scanner.cc:248:7: error: use of undeclared identifier 'memcpy'                                                                                                                                                                              
      memcpy((char*)&alignment_column, &buffer[offset], copied);                                                                                                                                                                                
      ^                                                                                                                                                                                                                                         
5 errors generated. 
```

--- a/README.md
+++ b/README.md
@@ -304,6 +304,7 @@ EOF
 
 You can also skip step 2 and use `:TSInstallFromGrammar zimbu` to install directly from a `grammar.js` in the top-level directory specified by `url`.
 Once the parser is installed, you can update it (from the latest revision of the `main` branch if `url` is a Github repository) with `:TSUpdate zimbu`.
+
 Note that neither `:TSInstall` nor `:TSInstallFromGrammar` copy query files from the grammar repository.
 If you want your installed grammar to be useful, you must manually [add query files](#adding-queries) to your local nvim-treesitter installation.
 Note also that nvim-treesitter modules are only triggered if your language's filetype is correctly identified.

--- a/README.md
+++ b/README.md
@@ -304,6 +304,11 @@ EOF
 
 You can also skip step 2 and use `:TSInstallFromGrammar zimbu` to install directly from a `grammar.js` in the top-level directory specified by `url`.
 Once the parser is installed, you can update it (from the latest revision of the `main` branch if `url` is a Github repository) with `:TSUpdate zimbu`.
+Note that neither `:TSInstall` nor `:TSInstallFromGrammar` copy query files from the grammar repository.
+If you want your installed grammar to be useful, you must manually [add query files](#adding-queries) to your local nvim-treesitter installation.
+Note also that nvim-treesitter modules are only triggered if your language's filetype is correctly identified.
+If neovim does not detect your language's filetype by default, you can add a short Vimscript file to nvim-treesitter's `ftdetect` runtime directory.
+See [Neovim's documentation](https://neovim.io/doc/user/filetype.html#new-filetype) on how to use Vimscript to detect a filetype.
 
 ## Update parsers used_by
 

--- a/README.md
+++ b/README.md
@@ -307,8 +307,8 @@ Once the parser is installed, you can update it (from the latest revision of the
 
 Note that neither `:TSInstall` nor `:TSInstallFromGrammar` copy query files from the grammar repository.
 If you want your installed grammar to be useful, you must manually [add query files](#adding-queries) to your local nvim-treesitter installation.
-Note also that nvim-treesitter modules are only triggered if your language's filetype is correctly identified.
-If neovim does not detect your language's filetype by default, you can add a short Vimscript file to nvim-treesitter's `ftdetect` runtime directory.
+Note also that module functionality is only triggered if your language's filetype is correctly identified.
+If Neovim does not detect your language's filetype by default, you can add a short Vimscript file to nvim-treesitter's `ftdetect` runtime directory.
 See [Neovim's documentation](https://neovim.io/doc/user/filetype.html#new-filetype) on how to use Vimscript to detect a filetype.
 
 ## Update parsers used_by
@@ -406,10 +406,17 @@ It can also help to update the parsers via `:TSUpdate`.
 
 #### Feature `X` does not work for `{language}`...
 
-First, check the `## {language} parser healthcheck` section of `:checkhealth` for any warning.
+First, check the `health#nvim_treesitter#check` and the `health#treesitter#check` sections of `:checkhealth` for any warning.
 If there is one, it's highly likely that this is the cause of the problem.
-If everything is okay, then it might be an actual error.
 
+Next check the `## Parser/Features` subsection of the `health#nvim_treesitter#check` section of `:checkhealth` to ensure the desired module is enabled for your language.
+If not, you might be missing query files; see [Adding queries](#adding-queries).
+
+Finally, ensure Neovim is correctly identifying your language's filetype using the `:echo &filetype` command while one of your language's files is open in Neovim.
+If not, add a short Vimscript file to nvim-treesitter's `ftdetect` runtime directory following [Neovim's documentation](https://neovim.io/doc/user/filetype.html#new-filetype) on filetype detection.
+You can also quickly & temporarily set the filetype for a single buffer with the `:set filetype=langname` command to test whether it fixes the problem.
+
+If everything is okay, then it might be an actual error.
 In that case, feel free to [open an issue here](https://github.com/nvim-treesitter/nvim-treesitter/issues/new/choose).
 
 #### I get `module 'vim.treesitter.query' not found`

